Tasks/Common/RemoteDeployer/RunPowerShellScriptJob.ps1

```
@@ -0,0 +1,122 @@
+$ExecutePsScript = {
+    Param(
+        [string] $scriptPath,
+        [string] $scriptArguments = "",
+        [string] $inlineScript,
+        [bool] $inline,
+        [string] $workingDirectory = "",
+        [string] $_errorActionPreference = "Continue",
+        [bool] $ignoreLASTEXITCODE,
+        [bool] $failOnStdErr,
+        [string] $initializationScriptPath,
+        [string] $sessionVariables
+    )
+
+    $Global:ErrorActionPreference = "Continue";
+
+    function Invoke-Tool {
+        Param(
+            [string] $toolPath,
+            [string] $toolArgs
+        )
+        $command = "& '$($toolPath.Replace('"', '').Replace("'", "''"))' $($toolArgs.Trim())"
+        Write-Host "##[command]$command"
+        Invoke-Expression $command
+    }
+
+    function Create-TemporaryFile {
+        Param(
+            [string] $contents
+        )
+        $filePath = [System.IO.Path]::Combine(([System.IO.Path]::GetTempPath()), ([guid]::NewGuid().ToString() + ".ps1"));
+        [System.IO.File]::WriteAllText(
+            $filePath,
+            $contents,
+            [System.Text.Encoding]::UTF8
+        )
+        return $filePath
+    }
+    
+    function Remove-TemporaryFile {
+        Param(
```

dont fail if unable to delete the file

log and silently continue